src/charm.py

```diff
@@ -149,76 +186,52 @@ def _check_and_report_k8s_conflict(self, error):
             return True
         return False
 
-    def _apply_k8s_resources(self, force_conflicts: bool = False) -> None:
-        """Applies K8S resources.
-
-        Args:
-            force_conflicts (bool): *(optional)* Will "force" apply requests causing conflicting
-                                    fields to change ownership to the field manager used in this
-                                    charm.
-                                    NOTE: This will only be used if initial regular apply() fails.
-        """
+    def _apply_k8s_resources(self, event: EventBase) -> None:
+        """Applies K8S resources."""
         self.unit.status = MaintenanceStatus("Creating K8S resources")
         try:
             self.crd_resource_handler.apply()
         except ApiError as error:
-            if self._check_and_report_k8s_conflict(error) and force_conflicts:
```

Comment:
Why aree we removing this one and just raising ? .... in the past we were facing issues.